hibernate-reactive-core/src/test/java/org/hibernate/reactive/NonNullableManyToOneTest.java

```diff
@@ -0,0 +1,154 @@
+/* Hibernate, Relational Persistence for Idiomatic Java
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * Copyright: Red Hat Inc. and Hibernate Authors
+ */
+package org.hibernate.reactive;
+
+import java.util.ArrayList;
+import java.util.List;
+import javax.persistence.Entity;
+import javax.persistence.Id;
+import javax.persistence.JoinColumn;
+import javax.persistence.ManyToOne;
+import javax.persistence.OneToMany;
+import javax.persistence.Table;
+
+import org.hibernate.cfg.Configuration;
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+import io.vertx.ext.unit.TestContext;
+
+public class NonNullableManyToOneTest extends BaseReactiveTest {
+
+	@Override
+	protected Configuration constructConfiguration() {
+		Configuration configuration = super.constructConfiguration();
+		configuration.addAnnotatedClass( Dealer.class );
+		configuration.addAnnotatedClass( Artist.class );
+		configuration.addAnnotatedClass( Painting.class );
+		return configuration;
+	}
+
+	@Before
+	public void populateDB(TestContext context) {
+		Artist artist = new Artist( "Grand Master Painter" );
+		artist.id = 1L;
+		Dealer dealer = new Dealer( "Dealer" );
+		dealer.id = 1L;
+		Painting painting = new Painting( "Mona Lisa");
+		painting.id = 2L;
+		artist.addPainting( painting );
+		dealer.addPainting( painting );
+
+		test( context, getMutinySessionFactory()
+				.withTransaction( s -> s.persistAll( painting, artist, dealer ) ) );
+	}
+
+	@After
+	public void cleanDB(TestContext context) {
+		test( context, deleteEntities( "Painting", "Artist" ) );
+	}
+
+	@Test
+	public void testNonNullableSuccess(TestContext context) {
+		test(
+				context,
+				getMutinySessionFactory().withTransaction( session -> session
+						.createQuery( "from Artist", Artist.class )
+						.getSingleResult().chain( a -> session.fetch( a.paintings ) )
+						.invoke( paintings -> {
+							context.assertNotNull( paintings );
+							context.assertEquals( 1, paintings.size() );
+							context.assertEquals( "Mona Lisa", paintings.get( 0 ).name );
+						} )
+						.chain( () -> getMutinySessionFactory().withTransaction( s -> session
+										.createQuery( "from Dealer", Dealer.class )
+										.getSingleResult().chain( d -> session.fetch( d.paintings ) )
+										.invoke( paintings -> {
+											context.assertNotNull( paintings );
+											context.assertEquals( 1, paintings.size() );
+											context.assertEquals( "Mona Lisa", paintings.get( 0 ).name );
```

Comment:
This indentation is weird. It makes it look like the second `withTransaction` is inside the first one (even if it's not the case).
It's not critical but since we have to wait for the next ORM release anyway, could you align them correctly, please? (Or maybe create two different methods)

Reply:
Actually, something is wrong in the stream.

Reply:
fixed alone with correcting sessions in test.